pkg/apis/core/v1alpha1/conversions.go

```diff
@@ -193,3 +193,112 @@ func Convert_v1alpha1_ShootStatus_To_core_ShootStatus(in *ShootStatus, out *core
 
 	return nil
 }
+
+func Convert_v1alpha1_ProjectSpec_To_core_ProjectSpec(in *ProjectSpec, out *core.ProjectSpec, s conversion.Scope) error {
+	if err := autoConvert_v1alpha1_ProjectSpec_To_core_ProjectSpec(in, out, s); err != nil {
+		return err
+	}
+
+	if owner := out.Owner; owner != nil {
+	outer:
+		for i, member := range out.Members {
+			if member.Name == owner.Name && member.APIGroup == owner.APIGroup && member.Kind == owner.Kind {
+				mustAddOwnerRole := true
+				for _, role := range member.Roles {
+					if role == core.ProjectMemberOwner {
+						mustAddOwnerRole = false
+						break outer
+					}
+				}
+
+				if mustAddOwnerRole {
+					out.Members[i].Roles = append(out.Members[i].Roles, core.ProjectMemberOwner)
+				}
+				continue
+			}
+
+			// delete owner role from all other members
+			for j, role := range member.Roles {
+				if role == ProjectMemberOwner {
+					out.Members[i].Roles = append(out.Members[i].Roles[:j], out.Members[i].Roles[j+1:]...)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func Convert_core_ProjectSpec_To_v1alpha1_ProjectSpec(in *core.ProjectSpec, out *ProjectSpec, s conversion.Scope) error {
+	if err := autoConvert_core_ProjectSpec_To_v1alpha1_ProjectSpec(in, out, s); err != nil {
+		return err
+	}
+
+	if owner := out.Owner; owner != nil {
+	outer:
+		for i, member := range out.Members {
+			if member.Name == owner.Name && member.APIGroup == owner.APIGroup && member.Kind == owner.Kind {
+				mustAddOwnerRole := true
+				for _, role := range member.Roles {
+					if role == ProjectMemberOwner {
+						mustAddOwnerRole = false
+						break outer
+					}
+				}
+
+				if mustAddOwnerRole {
+					out.Members[i].Roles = append(out.Members[i].Roles, ProjectMemberOwner)
+				}
+				continue
+			}
+
+			// delete owner role from all other members
+			for j, role := range member.Roles {
+				if role == ProjectMemberOwner {
+					out.Members[i].Roles = append(out.Members[i].Roles[:j], out.Members[i].Roles[j+1:]...)
+				}
+			}
+			if member.Role != nil && *member.Role == ProjectMemberOwner {
+				if len(out.Members[i].Roles) > 0 {
+					out.Members[i].Role = &out.Members[i].Roles[0]
+				} else {
+					out.Members[i].Role = nil
+				}
+			}
```

Comment:
At this point in time the member is not the correct owner. However, if it has `.spec.role=owner` set (maybe because it was the owner before) then we have to remove it now. We replace it with the first role in the `.spec.roles` list.

Reply:
Ah, got it 👍